test/lightning/projects/provisioner_test.exs

```diff
@@ -704,71 +785,175 @@ defmodule Lightning.Projects.ProvisionerTest do
         workflow: %{id: ^workflow_id}
       }
     end
+
+    test "audits workflow events as a result of the provisioner", %{
+      user: %{id: user_id} = user
+    } do
+      %{
+        body: body,
+        workflows: [
+          %{id: first_workflow_id},
+          %{id: second_workflow_id},
+          %{id: third_workflow_id}
+        ]
+      } = valid_document(nil, 3)
+
+      {:ok, project} =
+        Provisioner.import_document(%Lightning.Projects.Project{}, user, body)
+
+      {%{"id" => new_workflow_id} = new_workflow, _properties} =
+        valid_workflow(4)
+
+      body_with_modifications =
+        body
+        |> add_job_to_document(first_workflow_id, %{
+          "id" => Ecto.UUID.generate(),
+          "name" => "third-job",
+          "adaptor" => "@openfn/language-common@latest",
+          "body" => "console.log('hello world');"
+        })
+        |> remove_workflow_from_document(third_workflow_id)
+        |> add_workflow_to_document(new_workflow)
+
+      # Clear any audit records from the initial import
+      Repo.delete_all(Audit)
+
+      assert {:ok, _project} =
+               Provisioner.import_document(
+                 project,
+                 user,
+                 body_with_modifications
+               )
+
+      inserted_query =
+        from a in Audit,
+          where: a.item_id == ^new_workflow_id,
+          where: a.event == "inserted_by_provisioner",
+          where: a.actor_id == ^user_id
+
+      assert Repo.one(inserted_query) != nil
```

Comment:
Comparing against `nil` is generally not recommended. In this case you could just assert that it exists.

```suggestion
      assert Repo.exists?(inserted_query)
```

Reply:
@midigofrank Thanks for the advice - I have made the change in a new commit. The Internet does not provide a lot of useful data on why it is not recommended (outside of it being forbidden in Ecto) - are you able to provide any additional context as to why it is not recommended (if you don't have anything readily too hand, please don't spend a lot of time looking for the context - I am just curious).